src/Symfony/Contracts/Service/ServiceProviderInterface.php

```diff
@@ -39,7 +39,7 @@ public function has(string $id): bool;
      *  * ['foo' => '?'] means the container provides service name "foo" of unspecified type
      *  * ['bar' => '?Bar\Baz'] means the container provides a service "bar" of type Bar\Baz|null
      *
-     * @return string[] The provided service types, keyed by service names
+     * @return array<string, string> The provided service types, keyed by service names
```

Comment:
the key can be omitted which signals that the name should be the same as the type
maybe this is worth some more words in the description also?
```suggestion
     * @return array<string|int, string> The provided service types, keyed by service names
```

Reply:
`array<string|int, string>` is really the same than `string[]` (as array keys cannot be something else than `string|int`, so this would be the same than the current code.

However, the case of using an integer key to use the type as name is not documented in the phpdoc. This should be added if it is indeed supported.

Reply:
Indeed, in the examples showed in the PHPDoc, integer keys are not documented, and I don't think they should, neither the service provider should support it. What would be the use case? 
If you need a list of services you can use the `TaggedIterator`.

Reply:
The use case is omitting keys for brevity:

`$services[] = Foo::class` is the intepreted as `$services[Foo::class] = Foo::class`

Reply:
But this is just for the set of the internal array, when returning it it actually fills the array keys with a string in the `ServiceLocatorTrait`. Also, if you look at the consumers of this method on the Symfony codebase, they all expect a array key of string.

Reply:
@nicolas-grekas aren't you confusing ServiceProviderInterface with ServiceSubscriberInterface ?

Reply:
Yes indeed I am! Sorry for the confusion!